xdsl/xdsl_opt_main.py

```diff
@@ -267,6 +267,14 @@ def register_all_arguments(self, arg_parser: argparse.ArgumentParser):
             action="store_true",
             help="Allow the parsing of unregistered dialects.",
         )
+
+        arg_parser.add_argument(
```

Comment:
Same here, I would remove the argument, as we always want this option to be True.